src/core/memory.d

```diff
@@ -809,3 +809,113 @@ struct GC
         gc_runFinalizers( segment );
     }
 }
+
+/**
+ * Pure variants of C's memory allocation functions `malloc`, `calloc` and
+ * `realloc`. Purity is achieved via resetting the `errno` to it's value prior
+ * to being called, removing the function's global state mutation.
+ *
+ * See_Also:
+ *     $(LINK2 https://dlang.org/spec/function.html#pure-functions, D's rules for purity),
+ *     which allow for memory allocation under specific circumstances.
+ */
+void* pureMalloc(size_t size) @trusted pure @nogc nothrow
+{
+    const errno = fakePureGetErrno();
+    void* ret = fakePureMalloc(size);
+    if (!ret || errno != 0)
+    {
+        cast(void)fakePureSetErrno(errno);
+    }
+    return ret;
+}
+/// ditto
+void* pureCalloc(size_t nmemb, size_t size) @trusted pure @nogc nothrow
+{
+    const errno = fakePureGetErrno();
+    void* ret = fakePureCalloc(nmemb, size);
+    if (!ret || errno != 0)
+    {
+        cast(void)fakePureSetErrno(errno);
+    }
+    return ret;
+}
+/// ditto
+void* pureRealloc(void* ptr, size_t size) pure @nogc nothrow
+{
+    const errno = fakePureGetErrno();
+    void* ret = fakePureRealloc(ptr, size);
+    if (!ret || errno != 0)
+    {
+        cast(void)fakePureSetErrno(errno);
+    }
+    return ret;
+}
+
+/**
+ * Pure variant of C's memory deallocation function `free`.
+ */
+void pureFree(void* ptr) pure @nogc nothrow
+{
+    fakePureFree(ptr);
```

Comment:
Now I presume there's no need for the additional wrapper? Just rename fakePureFree to pureFree and move it here.